### PR TITLE
sol

### PR DESCRIPTION
**Pull request by Unlie9**

### CODE REVIEW

File: books/spiders/books.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.driver = webdriver.Chrome()
+
+    def close(self, reason: any) -> None:
+        self.driver.close()
+
+    def parse(self, response: Response, **kwargs) -> None:
```
Comment (DevSpaciX) [inline books/spiders/books.py]: Good job, but don't forget to use the correct type of annotations, It wouldn't be None

File: books/spiders/books.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        for book in response.css(".product_pod"):
+            detail_url = book.css("h3 > a::attr(href)").get()
+            yield response.follow(detail_url, callback=self.parse_book_info)
+
+    def parse_book_info(self, response: Response) -> dict:
```
Comment (DevSpaciX) [inline books/spiders/books.py]: the same, use the correct type of annotation. dict is wrong

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,55 @@
+attrs==24.2.0
```
Comment (DevSpaciX) [inline requirements.txt:1]: Why are there so many requirements, do you use them all?

**Review by DevSpaciX — CHANGES_REQUESTED** [review]
Great job! 
Please check comments

File: requirements.txt (hunk 3 above)
Comment (vladyslavchichkan) [reply requirements.txt:1]: made a freeze with interpreter, i dont wanna to remove because app can be able broken

**Review by DevSpaciX — APPROVED** [review]
Great job!

**Pull request closed without merging**